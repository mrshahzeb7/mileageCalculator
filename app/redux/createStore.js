import {applyMiddleware, compose, createStore} from 'redux';

export default (rootReducer, rootSaga) => {
  const middleware = [];
  const enhancers = [];

  enhancers.push(applyMiddleware(...middleware));

  const createAppropriateStore = createStore;

  const store = createAppropriateStore(rootReducer, compose(...enhancers));

  return {
    store,
  };
};
