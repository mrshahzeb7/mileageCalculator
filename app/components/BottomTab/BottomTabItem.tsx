import React from 'react';
import {
  ImageSourcePropType,
  StyleSheet,
  Text,
  TouchableOpacity,
} from 'react-native';
import {colors, fontSize, iconSizes, spacing} from '../../theme';
import {Icon} from '../Icon/Icon';
import {useAppSelector} from '../../redux';

export type BottomTabItemType = {
  id: number;
  icon: ImageSourcePropType;
  title: string;
  selected: boolean;
};

interface Props {
  item: BottomTabItemType;
  onPress: (item: BottomTabItemType) => void;
}

export const BottomTabItem: React.FC<Props> = ({item, onPress}) => {
  const selectedScreen = useAppSelector(
    state => state?.navigation?.selectedTabScreen,
  );

  const onItemPressed = () => {
    onPress?.(item);
  };

  const isSelectedTab = item.title === selectedScreen;
  return (
    <TouchableOpacity style={styles.container} onPress={onItemPressed}>
      <Icon
        style={[styles.icon, isSelectedTab && styles.selectedIcon]}
        size={iconSizes.small}
        icon={item.icon}
      />

      <Text style={[styles.title, isSelectedTab && styles.selectedTitle]}>
        {item.title}
      </Text>
    </TouchableOpacity>
  );
};

// styles
const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
  icon: {
    tintColor: colors.bottomTabIcon,
  },
  selectedIcon: {
    tintColor: colors.selectedBottomTabIcon,
  },
  title: {
    color: colors.bottomTabIcon,
    fontSize: fontSize.small,
    marginTop: spacing.vertical.tiny,
  },
  selectedTitle: {
    color: colors.selectedBottomTabIcon,
  },
});
