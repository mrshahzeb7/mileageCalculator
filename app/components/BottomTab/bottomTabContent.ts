import {icons} from '../../assets';
import {BottomTabItemType} from './BottomTabItem';

export type BottomTabScreensType = BottomTabItemType[];

export const bottomTabScreens: BottomTabScreensType = [
  {
    id: 1,
    icon: icons.Home,
    title: 'Home',
    selected: false,
  },
  {
    id: 2,
    icon: icons.TimeLine,
    title: 'TimeLine',
    selected: false,
  },
];
