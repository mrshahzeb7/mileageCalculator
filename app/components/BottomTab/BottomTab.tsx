import React, {useState} from 'react';
import {StyleSheet, View, ViewStyle} from 'react-native';
import {useSafeAreaInsets} from 'react-native-safe-area-context';
import {useDispatch} from 'react-redux';
import {NavigationActions} from '../../redux';
import {colors, spacing, styleGlobals} from '../../theme';
import {responsiveHeight} from '../../utils';
import {BottomTabItem, BottomTabItemType} from './BottomTabItem';
import {BottomTabScreensType, bottomTabScreens} from './bottomTabContent';

interface Props {}

export const BottomTab: React.FC<Props> = () => {
  const dispatch = useDispatch();

  const padding: ViewStyle = {
    paddingBottom: useSafeAreaInsets().bottom,
  };
  const [screens, setScreens] =
    useState<BottomTabScreensType>(bottomTabScreens);

  const onTabPressed = (selectedTab: BottomTabItemType) => {
    dispatch(NavigationActions.changeTabScreen(selectedTab.title));
    const newScreens = screens?.map(item => {
      item.selected = item.id === selectedTab?.id;
      return item;
    });

    setScreens(newScreens);
  };
  const renderItem = (item: BottomTabItemType) => {
    return <BottomTabItem key={item.id} item={item} onPress={onTabPressed} />;
  };
  return (
    <View style={[styles.container, padding]}>
      <View style={styles.itemWrapper}>{bottomTabScreens.map(renderItem)}</View>
    </View>
  );
};

// styles
const styles = StyleSheet.create({
  container: {
    ...styleGlobals.lowElevation,
    backgroundColor: colors.primary,
  },
  itemWrapper: {
    minHeight: responsiveHeight(8),
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    paddingVertical: spacing.vertical.tiny,
  },
});
