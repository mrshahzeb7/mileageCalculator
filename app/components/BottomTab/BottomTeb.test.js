import 'react-native';
import React from 'react';
import {BottomTab} from './BottomTab';
import renderer from 'react-test-renderer';
import {Provider} from 'react-redux';

import createStore from '../../redux';
import {
  SafeAreaProvider,
  initialWindowMetrics,
} from 'react-native-safe-area-context';

const store = createStore();

const Parent = () => {
  return (
    <SafeAreaProvider initialMetrics={initialWindowMetrics}>
      <Provider store={store}>
        <BottomTab />
      </Provider>
    </SafeAreaProvider>
  );
};

describe('BottomTab', () => {
  it(' BottomTab renders correctly', () => {
    const tree = renderer.create(<Parent />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
