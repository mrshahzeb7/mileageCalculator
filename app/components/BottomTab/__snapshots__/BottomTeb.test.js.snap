// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BottomTab  BottomTab renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    {
      "flex": 1,
    }
  }
/>
`;
